packages/cli/src/zosfiles/ZosFilesBase.handler.ts

```diff
@@ -39,33 +39,31 @@ export abstract class ZosFilesBaseHandler implements ICommandHandler {
      *
      * @returns {Promise<void>}
      */
-    public async process(commandParameters: IHandlerParameters) {
+    public async process(commandParameters: IHandlerParameters): Promise<void> {
         const sessCfg: ISession = ZosmfSession.createSessCfgFromArgs(
             commandParameters.arguments
         );
         const sessCfgWithCreds = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISession>(
-            sessCfg, commandParameters.arguments, {parms: commandParameters}
+            sessCfg, commandParameters.arguments, { parms: commandParameters }
         );
-
         const session = new Session(sessCfgWithCreds);
-        const response = await this.processWithSession(commandParameters, session);
-
         commandParameters.response.progress.endBar(); // end any progress bars
-        // Print out the response
-        if (response.commandResponse) {
-            commandParameters.response.console.log(response.commandResponse);
-        }
-
-        // Return as an object when using --response-format-json
-        commandParameters.response.data.setObj(response);
-
-        // Ensure error gets thrown if request was unsuccessful.
-        // Sometimes it is useful to delay throwing an error until the end of the handler is
-        // reached, for example the upload API needs to return an API response even when it fails.
-        if (!response.success && response.commandResponse) {
-            throw new ImperativeError({
-                msg: response.errorMessage || response.commandResponse
-            });
+        try {
+            const response = await this.processWithSession(commandParameters, session);
+            if (response.commandResponse) {
+                commandParameters.response.console.log(response.commandResponse);
+            }
+            commandParameters.response.data.setObj(response);
+        } catch (error) {
+            if (commandParameters.arguments.ignoreNotFound && (error.errorCode === '404' || error.toString().includes("IDC3012I "))) {
+                // The IDC3012I code is an IBM z/OS error message that indicates that the
+                // requested dataset or VSAM entry does not exist in the catalog
+                commandParameters.response.data.setObj({ success: true });
+            } else {
+                throw new ImperativeError({
+                    msg: error.mMessage
+                });
```

Comment:
This is fine! 🔥 🐶 🔥 

What about the processWithSession implementations that do not throw an error, but instead return "gracefully" with `success: false` and a proper response on the commandResponse object?

Even though the ZosFilesBaseHandler is [not exported to end users/developers](https://github.com/zowe/zowe-cli/blob/next/packages/cli/src/index.ts), we still need to maintain the behavior for our own APIs.

Here are the ones that I could find quickly:
- List.dataSetsMatchingPattern
- Download.allMembers
- Upload.pathToDataSet

I believe we still need to check for the response.success value and throw and error (in the `try` block)

Reply:
Changes to this file will affect every command in the `zowe zos-files` group. If the `IDC3012I` error is specific to VSAMs, would it make sense to handle it in `delete/vsam/Vsam.handler.ts` instead?